modules/redis/src/main/scala/zio/redis/internal/SingleNodeExecutor.scala

```diff
@@ -62,23 +63,42 @@ private[redis] final class SingleNodeExecutor private (
 }
 
 private[redis] object SingleNodeExecutor {
-  lazy val layer: ZLayer[RedisConfig, RedisError.IOError, RedisExecutor] =
+  lazy val layer: ZLayer[RedisConfig, RedisError.IOError, RedisExecutor[RedisExecutor.Sync]
+    & RedisExecutor[RedisExecutor.Async]] =
     RedisConnection.layer >>> makeLayer
 
-  lazy val local: ZLayer[Any, RedisError.IOError, RedisExecutor] =
+  lazy val local
+    : ZLayer[Any, RedisError.IOError, RedisExecutor[RedisExecutor.Sync] & RedisExecutor[RedisExecutor.Async]] =
     RedisConnection.local >>> makeLayer
 
-  def create(connection: RedisConnection): URIO[Scope, SingleNodeExecutor] =
+  def create(
+    connection: RedisConnection
+  ): URIO[Scope, ZEnvironment[SingleNodeExecutor[RedisExecutor.Sync] & SingleNodeExecutor[RedisExecutor.Async]]] =
     for {
-      requests  <- Queue.bounded[Request](RequestQueueSize)
-      responses <- Queue.unbounded[Promise[RedisError, RespValue]]
-      executor   = new SingleNodeExecutor(connection, requests, responses)
-      _         <- executor.run.forkScoped
-      _         <- logScopeFinalizer(s"$executor Node Executor is closed")
-    } yield executor
+      requests     <- Queue.bounded[Request](RequestQueueSize)
+      responses    <- Queue.unbounded[Promise[RedisError, RespValue]]
+      executor      = new SingleNodeExecutor(
+                        connection,
+                        requests,
+                        responses,
+                        new ToG[RedisExecutor.Sync] { def apply[A](r: UIO[IO[RedisError, A]]) = r.flatten }
+                      )
+      asyncExecutor = new SingleNodeExecutor(
+                        connection,
+                        requests,
+                        responses,
+                        new ToG[RedisExecutor.Async] { def apply[A](r: UIO[IO[RedisError, A]]) = r }
+                      )
+      _            <- executor.run.forkScoped
```

Comment:
We create both, run the "flattening" one, and extract the appropriate based on user's request (i.e., `makeLayer`), right?

I'm thinking whether it's possible to unify them somehow. If I understand the code correctly, "flattening" happens only in `RedisCommand` (cluster executor might be an exception to this, not sure about pub-sub).

Reply:
Right, while thinking about your comment, I realized that it's actually not necessary to have a `G[_]` parameter in `RedisExecutor` at all, and also in `RedisCommand` it's not really required. So I have now pushed a significantly simplified version that only creates a single Executor.